### PR TITLE
[Settings] Use a custom serializer instead of relying upon the YAML module

### PR DESCRIPTION
**Pull request by segiddins**
¯\\\_(ツ)\_/¯ seems like a good idea

\c @RochesterinNYC @indirect

### CODE REVIEW

**Comment by segiddins** [comment]
W00t this passes!

File: lib/bundler/settings.rb [hunk 1]
```diff
@@ -201,14 +201,21 @@ def set_key(key, value, hash, file)
         hash.delete(key) if value.nil?
         SharedHelpers.filesystem_access(file) do |p|
           FileUtils.mkdir_p(p.dirname)
-          require "bundler/psyched_yaml"
```
Comment (RochesterinNYC) [inline lib/bundler/settings.rb]: ooo this is the start of something beautiful.

File: lib/bundler/settings.rb [hunk 2]
```diff
@@ -201,14 +201,21 @@ def set_key(key, value, hash, file)
...
         end
       end
 
       value
     end
 
+    def serialize_hash(hash)
+      yaml = String.new("---\n")
+      hash.each do |key, value|
+        yaml << key << ": " << value.to_s.gsub(/\s+/, " ").inspect << "\n"
```
Comment (RochesterinNYC) [inline lib/bundler/settings.rb]: There aren't any current cases where the `.bundle/config` would have or require something more complex than just simple key --> value mapping is there?

Comment (segiddins) [inline lib/bundler/settings.rb]: it can't read anything else, so no?

**Comment by segiddins** [comment]
@indirect r?

**Comment by segiddins** [comment]
@indirect ping

**Comment by segiddins** [comment]
@homu r+

**Pull request merged**